test/source/mock/google-api-mock.ts

```diff
@@ -190,9 +190,28 @@ const parseMultipartDataAsMimeMsg = async (multipartData: string): Promise<Parse
   return parsed;
 };
 
-const validateMimeMsg = (acct: string, mimeMsg: ParsedMail, threadId?: string) => {
-  if (threadId && !new Data(acct).getThreads().find(t => t.id === threadId)) {
-    throw new HttpClientErr('The thread you are replying to not found', 404);
+const validateMimeMsg = async (acct: string, mimeMsg: ParsedMail, threadId?: string) => {
+  if (threadId) {
+    const messages = new Data(acct).getMessagesByThread(threadId);
+    if (!messages || !messages.length) {
+      throw new HttpClientErr('The thread you are replying to not found', 404);
+    }
+    const inReplyToMessageId = mimeMsg.headers.get('in-reply-to') ? mimeMsg.headers.get('in-reply-to')!.toString() : '';
+    if (inReplyToMessageId) {
+      let isMessageExists = false;
+      for (const message of messages) {
+        if (message.raw) {
+          const parsedMimeMsg = await Parse.convertBase64ToMimeMsg(message.raw);
+          if (parsedMimeMsg.messageId === inReplyToMessageId) {
+            isMessageExists = true;
+            break;
+          }
+        }
+      }
+      if (!isMessageExists) {
+        throw new HttpClientErr('You are replying to unexisting message id.', 400);
```

Comment:
Please make the error message a bit more verbose (it's for internal debugging, so good to know exactly what's up)
```
Error: suplied In-Reply-To header (${inReplyToMessageId}) does not match any messages present in the mock data for thread ${threadId}
```

Reply:
Okay, that looks better